src/main/kotlin/app/revanced/patches/youtube/layout/hidetimeandseekbar/patch/HideTimeAndSeekbarPatch.kt

```diff
@@ -0,0 +1,72 @@
+package app.revanced.patches.youtube.layout.hidetimeandseekbar.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.youtube.layout.hidetimeandseekbar.fingerprints.InlineTimeBarWrapperFingerprint
+import app.revanced.patches.youtube.layout.hidetimeandseekbar.fingerprints.TimeCounterFingerprint
+import app.revanced.patches.youtube.layout.hidetimeandseekbar.annotations.HideTimeAndSeekbarCompatibility
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("e-sport-mode")
+@Description("Hides progress bar and time counter on videos.")
+@HideTimeAndSeekbarCompatibility
+@Version("0.0.1")
+class HideTimeAndSeekbarPatch : BytecodePatch(
+    listOf(
+        InlineTimeBarWrapperFingerprint, TimeCounterFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_hide_time_and_seekbar",
+                StringResource("revanced_hide_time_and_seekbar_title", "Hide time and seekbar"),
+                false,
+                StringResource("revanced_hide_time_and_seekbar_summary_on", "Time and seekbar are hidden"),
+                StringResource("revanced_hide_time_and_seekbar_summary_off", "Time and seekbar are visible")
+            )
+        )
+
+        val inlineTimeBarWrapperMethod = InlineTimeBarWrapperFingerprint.result!!.mutableMethod
+
+        inlineTimeBarWrapperMethod.addInstructions(
+            0, """
+            invoke-static { }, Lapp/revanced/integrations/patches/HideTimeAndSeekbarPatch;->hideTimeAndSeekbar()Z
```

Comment:
The integrations code for this is missing, please do not forget PRing this before marking this PR as ready.